src/redux/api/slice.js

```diff
@@ -0,0 +1,46 @@
+import { createApi } from '@reduxjs/toolkit/query/react';
+import { baseQueryWithReauth } from './baseQuery';
+
+// Define our single API slice object
+export const apiSlice = createApi({
+  reducerPath: 'api',
+  baseQuery: baseQueryWithReauth,
+  // The "endpoints" represent operations and requests for this server
+  // nodeId is passed in JSON-LD '@id' key, https://www.w3.org/TR/2014/REC-json-ld-20140116/#node-identifiers
+  endpoints: builder => ({
+    subscriptionGenerateOrders: builder.mutation({
+      query: date => ({
+        url: 'api/recurrence_rules/generate_orders',
+        params: {
+          date: date.format('YYYY-MM-DD'),
+        },
+        method: 'POST',
+        body: {},
+      }),
+    }),
+    getMyTasks: builder.query({
+      query: date => `api/me/tasks/${date.format('YYYY-MM-DD')}`,
+    }),
+    getOrderTiming: builder.query({
+      query: nodeId => `${nodeId}/timing`,
+    }),
+    getOrderValidate: builder.query({
+      query: nodeId => `${nodeId}/validate`,
+    }),
+    updateOrder: builder.mutation({
+      query: ({ nodeId, ...patch }) => ({
+        url: nodeId,
+        method: 'PUT',
+        body: patch,
+      }),
```

Comment:
I am surprised by the functions that are not `getMyTasks`
1. they are not linked to the PR
2. they are not explictly called? or maybe they will be called on onFocus/onOnline... aso, but it seems weird to me for example for `updateOrder`?
3. we should build apiSlice differently according to the user roles? (getMytasks is for courier, generateOrders for dispatchers, updateorder for admin/restaurant owner ?

Reply:
It's actually a copy-paste of https://github.com/coopcycle/coopcycle-web/blob/master/js/app/api/slice.js with `getMyTasks` added. I mentioned elsewhere that, ideally, I see it in a library, but for now, I think we should try to make them compatible so it's easy to copy-paste already implemented endpoints between the web and the app

1. not directly
2. no, they are not used at the moment, but will be soon. In a timing modal, for example
3. I thought about it, but rkt query actually advises against it (see Tip): https://redux-toolkit.js.org/rtk-query/api/createApi, so I decided to stick to the standard approach

Reply:
3. i am not sure we understand the same, i was thinking to generate dynamically the slice upon login so you don't load the automatic refetch for tasks for a customer for example

What would happen with the current code if a simple customer logs in for example? will it try to refetch tasks (hence 401) on focus? I can test that later

Reply:
No, at least as I understand. Refetch happens only in `TasksPage` component https://github.com/coopcycle/coopcycle-app/blob/ad73ae221e776a2c2e6706426c0a4751fe8179ca/src/navigation/courier/TasksPage.js#L58 Until you reach it no refetch is setup. 

`setupListenersReactNative` https://github.com/coopcycle/coopcycle-app/blob/f7667c26a9c822427f6ef30aea845baf59f7436a/src/redux/setupListenersReactNative.js only tell rtk about the state changes, but they don't trigger any requests until they are configured elsewhere